iceoryx2-bb/lock-free/src/mpmc/unique_index_set.rs

```diff
@@ -201,20 +231,47 @@ impl Drop for UniqueIndex<'_> {
 pub struct UniqueIndexSet {
     data_ptr: RelocatablePointer<UnsafeCell<u32>>,
     capacity: u32,
-    borrowed_indices: IoxAtomicUsize,
     pub(crate) head: IoxAtomicU64,
     is_memory_initialized: IoxAtomicBool,
 }
 
 unsafe impl Sync for UniqueIndexSet {}
 unsafe impl Send for UniqueIndexSet {}
 
+const LOCK_ACQUIRE: u32 = 0x00ffffff;
+
+struct HeadDetails {
+    head: u32,
+    aba: u16,
+    borrowed_indices: u32,
+}
+
+impl HeadDetails {
+    fn from(value: u64) -> Self {
+        Self {
+            head: ((value & 0xffffff0000000000) >> 40) as u32,
+            aba: ((value & 0x000000ffff000000) >> 24) as u16,
+            borrowed_indices: (value & 0x0000000000ffffff) as u32,
```

Comment:
This is more common and less error prone
```suggestion
            head: ((value >> 40) as u32) & 0xFFFFFF,
            aba: ((value >> 24) as u16,
            borrowed_indices: (value as u32) & 0xFFFFFF,
```
For head it could be even just `head: (value >> 40) as u32` but the compiler will optimize that away